### PR TITLE
Fix/3722 Add confirm popup to invalid mentions

### PR DESCRIPTION
**Pull request by gbonaspetti**
#3722 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] If there are modifications in the Tracim configuration files (eg. `development.ini`), make sure it is documented in `backend/doc/setting.md` and notify relevant people of this change

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend_lib/test/mention.spec.js [hunk 1]
```diff
@@ -231,4 +239,12 @@ describe('mention.js', () => {
       })
     })
   })
+
+  describe('getInvalidMentionsList()', () => {
+    it('should return the invalid mentions in content', () => {
+      const knownMentions = [{ mention: 'user1' }, { mention: 'user2' }]
+      const content = '<p>This is a content with two @user1 @user1 mentions and a @invalid_mention also two @not_a_member @not_a_member mentions</p>'
```
Comment (raphj) [inline frontend_lib/test/mention.spec.js:246]:
```suggestion
      const content = '<p>This is a content with two @user1 @user1 mentions and a @invalid_mention also two @not_a_member @not_a_member mentions, and also group mentions @all @tous @todos :-)</p>'
```

File: frontend_app_thread/src/container/Thread.jsx [hunk 2]
```diff
@@ -285,7 +288,25 @@ export class Thread extends React.Component {
...
+
+    if (invalidMentionList.length > 0) {
+      this.setState({
+        invalidMentionList: invalidMentionList,
+        showInvalidMentionPopupInComment: true
+      })
+    } else this.handleClickValidateAnywayNewComment()
+  }
+
+  handleClickValidateAnywayNewComment = () => {
```
Comment (raphj) [inline frontend_app_thread/src/container/Thread.jsx:309]: Maybe put this duplicated code in frontend lib?

**Review by raphj — CHANGES_REQUESTED** [review]

File: frontend_lib/src/mention.js [hunk 3]
```diff
@@ -47,7 +53,14 @@ const isAWrappedMention = (node) => (
   node.id.startsWith(MENTION_ID_PREFIX)
 )
 
-export const wrapMentionsInSpanTags = (node, doc) => {
+export const getInvalidMentionsList = (content, knownMentions) => {
```
[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend_lib/src/mention.js]: Changed.

File: frontend_lib/test/mention.spec.js (hunk 1 above)
Comment (gbonaspetti) [reply frontend_lib/test/mention.spec.js]: Changed.

File: frontend_lib/src/mention.js [hunk 4]
```diff
@@ -47,7 +53,14 @@ const isAWrappedMention = (node) => (
...
+  const doc = getDocumentFromHTMLString(content)
+  let mentionsInContent = doc.body.textContent.match(MENTION_REGEX_GLOBAL) || []
+  mentionsInContent = [...new Set(mentionsInContent)]
+  return mentionsInContent.filter(mention => knownMentions.indexOf(mention) === -1)
```
[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend_lib/src/mention.js]: @raphj did not work, but I did another version based on yours

File: frontend_app_thread/src/container/Thread.jsx (hunk 2 above)
Comment (gbonaspetti) [reply frontend_app_thread/src/container/Thread.jsx]: Changed.

**Review by raphj — APPROVED** [review]

File: frontend_lib/src/mention.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (raphj) [reply frontend_lib/src/mention.js]: Ah yes, there is an `@` sign in the mention, I had forgotten :-P

Good for me.

**Comment by gbonaspetti** [comment]
Frontend part finished and approved, missing the backend part to merge (issue will return to To Do).

**Pull request merged**